### ISSUE TITLE
Save State function randomly loads old savestate that shouldn't exist anymore

### ISSUE THREAD

**Issue by zant0** [loganmc10/m64p#80]
I'm using the save state function by pressing F5 to save and F7 to load. After playing for some time while often making save states and loading them, the emulator loads not the last savestate I made but an old one I made like ages ago. The old save state the emulator then loads isn't even from the same game session. It can be one from yesterday. The bug isn't reproducible in the sense that I could give you a step by step instruction in how to make the emulator load an old savestate but it occurred several times.

OS: Windows 10 64 Bit
Game: Super Mario 64 NTSC-U
m64p build: Oct 27 2020 (but it also happend in earlier builds...)

**Comment by Jj0YzL5nvJ** [comment]
Try:
`Windows+R` and execute `explorer "%LOCALAPPDATA%\VirtualStore".`

Delete all the duplicates and everything that you don't know what it's.
Cleaning temp's and doing `chkdsk` regularly (every 3 months) can do wonders in your case.

**Comment by loganmc10** [comment]
@zant0 what might be happens is that you might be changing the save state slot without realizing it.

Mupen64Plus has these built-in keyboard shortcuts:
![image](https://user-images.githubusercontent.com/848146/97444860-38db9900-18f2-11eb-837f-16067fb2670c.png)

Pressing 0-9 will change the save state "slot", so basically if you press 0, you're in slot "0", and pressing F5/F7 will save & load to that slot. If you press "1", now you're dealing with different save states.

So basically, if you press "0", then F5, you've saved to slot 0. If you then change to slot 1 and hit F5, you'll save a different save state. If you hit "0" again and hit F7, you'll load the first save state.

Mupen64plus defaults to slot 0 on launch

**Comment by zant0** [comment]
@loganmc10 thats it. going through the slots I found other save states. 

What I found is that if you change the save slot with the keyboard shortcuts under "File -> Change Save Slot" it doesn't indicate that. When restarting the emulator save slot 0 is shown to be the active save slot under "File -> Change Save Slot" while the actual active save slot is the last one you have selected either with the hotkeys or over the menu bar.

**Comment by loganmc10** [comment]
> When restarting the emulator save slot 0 is shown to be the active save slot under "File -> Change Save Slot" while the actual active save slot is the last one you have selected either with the hotkeys or over the menu bar.

I did notice did, I just fixed it today and it'll be included in the next release. When you start the emulator, the accurate save slot will be indicated in that drop down.

> What I found is that if you change the save slot with the keyboard shortcuts under "File -> Change Save Slot" it doesn't indicate that.

This is a bit trickier to fix. The emulator doesn't send any feedback to the GUI saying "I changed the save slot!", so I don't know how I would update the GUI to reflect that. With the fix I mentioned above, the GUI will be accurate if you close/reopen the emulator, but not if you change the slot using a keyboard shortcut ingame
